File: AVc/app/src/main/java/com/hfad/AVc/ui/database/ContactDao.java
package com.hfad.avc.ui.database;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.provider.ContactsContract;
import android.util.Log;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.hfad.avc.Applications;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

import io.reactivex.rxjava3.core.Observable;

@Dao
public interface ContactDao {
    @Query("SELECT * FROM contact")
    List<Contact> getAll();

    @Query("SELECT * FROM contact WHERE (favorite != 0)")
    List<Contact> getComingCongratulations();

    @Query("SELECT * FROM contact WHERE favorite != 0")
    List<Contact> getAllCongratulations();

    @Query("SELECT * FROM contact WHERE id = :id")
    Contact getById(String id);

    @Insert
    void insert(Contact contact);

    @Update
    void update(Contact contact);

    @Delete
    void delete(Contact contact);



}

File: app/src/main/java/com/lastaurus/automatic_congratulations/data/database/ContactDao.kt
package com.lastaurus.automatic_congratulations.data.database

import androidx.room.*
import com.lastaurus.automatic_congratulations.data.model.Contact
import kotlinx.coroutines.flow.Flow

@Dao
interface ContactDao {

   @get:Query("SELECT * FROM contact ORDER BY name")
   val all: Flow<List<Contact>>

   @get:Query("SELECT * FROM contact ORDER BY name DESC")
   val allDESC: Flow<List<Contact>>


   @get:Query("SELECT * FROM contact WHERE (favorite != 0) ORDER BY name")
   val favorite: Flow<List<Contact>>

   @get:Query("SELECT * FROM contact WHERE (favorite != 0) ORDER BY name DESC")
   val favoriteDESC: Flow<List<Contact>>

   @Query("SELECT * FROM contact WHERE id = :id")
   fun getById(id: Int): Contact?

   @Query("SELECT * FROM contact WHERE idInBase = :idInBase")
   fun getByIdInBase(idInBase: String): Contact?

   @Query("SELECT COUNT(*)>0 FROM contact WHERE idInBase = :idInBase")
   fun contactInBase(idInBase: String): Boolean

   @Query("SELECT * FROM contact WHERE name = :name")
   fun getByName(name: String): Contact?

   @get:Query("SELECT COUNT(*) FROM contact")
   val size: Int

   @Query("SELECT * FROM contact WHERE name LIKE '%' || :searchText || '%' ORDER BY name")
   fun searchContact(searchText: String): Flow<List<Contact>>

   @Insert
   fun insert(contact: Contact)

   @Update
   fun update(contact: Contact)

   @Upsert
   fun upsert(contact: Contact)

   @Delete
   fun delete(contact: Contact)
}